### PR TITLE
Set openfile params when following internal links

### PR DESCRIPTION
**Pull request by artonge**
Extend https://github.com/nextcloud/server/pull/27102 to internal links to improve share experience.

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]
Nice :+1:

**Comment by artonge** [comment]
@dependabot-bot rebase

**Comment by artonge** [comment]
@dependabot-bot rebase

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
backport?

File: apps/files/lib/Controller/ViewController.php [hunk 1]
```diff
@@ -391,10 +391,11 @@ private function provideInitialState(string $dir, ?string $openfile): void {
 	 * Redirects to the file list and highlight the given file id
 	 *
 	 * @param string $fileId file id to show
+	 * @param bool $setOpenfile - wether or not to set the openfile URL parameter
```
Comment (Pytal) [inline apps/files/lib/Controller/ViewController.php:394]:
```suggestion
	 * @param bool $setOpenfile - whether or not to set the openfile URL parameter
```

Comment (artonge) [reply apps/files/lib/Controller/ViewController.php]: Nice catch, thanks :+1:

**Pull request merged**